pkg/plugin/validation.go

```diff
@@ -17,32 +17,64 @@ limitations under the License.
 package plugin
 
```

Comment:
the name of the file is validation, so should we not centralize here all plugin validations? 
Or your idea is just to have the plugin version validations? If yes, could we rename the file to make it clear or centralize all plugin validations?

Reply:
This file contains functions that let you build a plugin validator by defining what a valid plugin name/version is. I don't see a reason to change anything but error types.